artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -168,7 +160,8 @@ func consumeDelayedArtifactsFiles(pcWrapper *producerConsumerWrapper, filesToCon
 			return err
 		}
 
-		shouldStop, err := uploadByChunks(delayedArtifactsFile.DelayedArtifacts, uploadChunkChan, base, delayHelper, errorsChannelMng, pcWrapper)
+		sortedDelayedArtifacts := sortDelayedArtifacts(delayedArtifactsFile, delayHelper)
```

Comment:
I believe this is only sorting the artifacts within a specific delayedArtifactsFile. Those files are handled in parallel, meaning that even if a single file is being handled in the correct order, artifacts from different files can be uploaded in the wrong order.

Reply:
Good catch! Thanks